app/src/main/java/org/glucosio/android/activity/AppLockActivity.java

```diff
@@ -0,0 +1,54 @@
+package org.glucosio.android.activity;
+
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.util.Log;
+
+import org.glucosio.android.R;
+import org.glucosio.android.tools.TimeElapsed;
+
+import java.util.Calendar;
+
+/**
+ * Created by Viney Ugave on 10/25/16.
+ */
+public class  AppLockActivity extends AppCompatActivity {
+
+    private static final String TAG  = "AppLockActivity";
+    private static TimeElapsed timeElapsedInstance;
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        timeElapsedInstance = TimeElapsed.getInstance();
+        timeElapsedInstance.resetTime();
+        timeElapsedInstance.setStartTime();
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        //check if app has passed a time threshold
+        if(timeElapsedInstance.getStartTime() != 0){
+            timeElapsedInstance.setEndTime(Calendar.getInstance().getTimeInMillis());
+            long threshold = timeElapsedInstance.getEndTime()-timeElapsedInstance.getStartTime();
+            Log.d(TAG,"Threshold : "+threshold);
+            //Current timeout threshold set to 30s
+            if(threshold>30000){
```

Comment:
This is wrong algorithm. Every resume of the activity that happens after 30 seconds after create will "lock" it. Basically, it will happen every dialog message showing or minimise/maximise app.